hermes-management/src/main/java/pl/allegro/tech/hermes/management/config/storage/StorageConfiguration.java

```diff
@@ -0,0 +1,167 @@
+    public SharedCounter sharedCounter() {
+        CuratorFramework curatorFramework = clientManager().getLocalClient().getCuratorFramework();
+        return new SharedCounter(curatorFramework,
+                storageClustersProperties.getSharedCountersExpiration(),
+                storageClustersProperties.getRetrySleep(),
+                storageClustersProperties.getRetryTimes());
+    }
+
+    @Bean
+    public DistributedEphemeralCounter distributedCounter() {
+        CuratorFramework curatorFramework = clientManager().getLocalClient().getCuratorFramework();
+        return new DistributedEphemeralCounter(curatorFramework);
+    }
+
+    @Bean
+    GroupRepository groupRepository() {
+        ZookeeperClient localClient = clientManager().getLocalClient();
+        return new ZookeeperGroupRepository(localClient.getCuratorFramework(), objectMapper, zookeeperPaths());
+    }
+
+    @Bean
+    TopicRepository topicRepository() {
+        ZookeeperClient localClient = clientManager().getLocalClient();
+        return new ZookeeperTopicRepository(localClient.getCuratorFramework(), objectMapper, zookeeperPaths(),
+                groupRepository());
+    }
+
+    @Bean
+    SubscriptionRepository subscriptionRepository() {
+        ZookeeperClient localClient = clientManager().getLocalClient();
+        return new ZookeeperSubscriptionRepository(localClient.getCuratorFramework(), objectMapper, zookeeperPaths(),
+                topicRepository());
+    }
+
+    @Bean
+    OAuthProviderRepository oAuthProviderRepository() {
+        ZookeeperClient localClient = clientManager().getLocalClient();
+        return new ZookeeperOAuthProviderRepository(localClient.getCuratorFramework(), objectMapper, zookeeperPaths());
+    }
+
+    @Bean
+    MessagePreviewRepository messagePreviewRepository() {
+        ZookeeperClient localClient = clientManager().getLocalClient();
+        return new ZookeeperMessagePreviewRepository(localClient.getCuratorFramework(), objectMapper, zookeeperPaths());
+    }
+
+    @Bean
+    TopicBlacklistRepository topicBlacklistRepository() {
+        ZookeeperClient localClient = clientManager().getLocalClient();
+        return new ZookeeperTopicBlacklistRepository(localClient.getCuratorFramework(), objectMapper, zookeeperPaths());
+    }
+
+    @Bean
+    SubscriptionOffsetChangeIndicator subscriptionOffsetChangeIndicator() {
+        ZookeeperClient localClient = clientManager().getLocalClient();
+        return new ZookeeperSubscriptionOffsetChangeIndicator(localClient.getCuratorFramework(), zookeeperPaths(),
+                subscriptionRepository());
+    }
+
+    @Bean
+    AdminTool adminTool() {
+        ZookeeperClient localClient = clientManager().getLocalClient();
+        return new ZookeeperAdminTool(zookeeperPaths(), localClient.getCuratorFramework(),
+                objectMapper, Configs.ADMIN_REAPER_INTERAL_MS.getDefaultValue());
+    }
+
+    @Bean
+    UndeliveredMessageLog undeliveredMessageLog() {
```

Comment:
We don’t need this bean anymore as it stopped being used. Now we have ZookeeperUndeliveredMessageLog per cluster.